lru_cache/lru_cache.py

```diff
@@ -29,5 +67,40 @@ def get(self, key):
     want to overwrite the old value associated with the key with
     the newly-specified value.
     """
+    #set first so we have something to get
     def set(self, key, value):
-        pass
+        # need to access the key to remove it later
+        if key in self.storage:
+            node = self.storage[key]
+            node.value = (key, value)
+            self.order.move_to_front(node)
+            return
+        else:
+            if self.size == self.limit:
+                #shouldnt be greater than capacity
+                del self.storage[self.order.tail.value[0]]
+                self.order.remove_from_tail() #LRU
+                self.size -= 1
+
```

Comment:
As long as you are consistent, you can treat either the head or the tail as the "top" or the latest accessed item!